spring-cloud-task-samples/tasksink/src/test/java/io/spring/TaskSinkApplicationTests.java

```diff
@@ -0,0 +1,65 @@
+package io.spring;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.spring.configuration.TaskSinkConfiguration;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.deployer.spi.task.LaunchState;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.task.launcher.TaskLaunchRequest;
+import org.springframework.cloud.task.launcher.TaskLauncherSink;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.messaging.support.GenericMessage;
+
+/**
+ * @author Glenn Renfro
+ */
+public class TaskSinkApplicationTests {
+
+	private AnnotationConfigApplicationContext context;
+
+	@Before
+	public void setUp() {
+		context = new AnnotationConfigApplicationContext();
+		context.setId("sinkTest");
+	}
+
+	@After
+	public void tearDown(){
+		if(context != null){
+			context.close();
+		}
+	}
+
+	@Autowired
+	@Bindings(TaskLauncherSink.class)
+	private Sink sink;
+
+	@Test
+	public void testLaunch() {
+		context.register(TaskSinkApplication.class, TaskSinkConfiguration.class);
+		context.refresh();
```

Comment:
Is there a reason to do this within the test instead of using the Spring runner and `@ContextConfiguration`?